hw/xbox/xbox.c

```diff
@@ -502,6 +518,11 @@ static inline void xbox_machine_initfn(Object *obj)
     object_property_set_description(obj, "eeprom",
                                     "Xbox EEPROM file", NULL);
 
+    object_property_add_str(obj, "avpack", machine_get_avpack,
+                            machine_set_avpack, NULL);
+    object_property_set_description(obj, "avpack",
+                                    "Xbox video connector: composite (default), scart, svideo, vga, vga-sog, hdtv", NULL);
```

Comment:
Missing a `none` option (for when no cable is plugged in). Also see my comment regarding MS names for cable types.

Reply:
@JayFoxRox there is no definition for "none" in the xqemu code, at least I didn't found it.

Used sources:
- https://github.com/xqemu/xqemu/blob/3b29f0888a2f942b9ad91907e4e12d47a72049f8/hw/xbox/smbus_xbox_smc.c#L55
- https://github.com/XboxDev/cromwell/blob/5b06e83cd1e2b0f7587e06946ffcc0f587536cc0/drivers/video/VideoInitialization.h#L15

I'm suspecting `0xFF` value, but there are no sources confirming that.

Reply:
The 3 mode pin bits will be 1 when no cable is plugged in. So the SMC constant is 7. See link from mentioned comment: https://xboxdevwiki.net/AV_Cables

Ditch the current names (fabricated / guessed or by homebew scene / xbox-linux) and use the MS names.

MS never even released a VGA cable for Xbox, let alone the VGA-SOG stuff. Some modded bios added support for these modes, but there's no official adapters. We should focus on official / licensed original Xbox stuff, not some home-made cables which require hacks.

The only unknown cable type is SMC constant 5, which the kernel treats largely the same as mode 7 (no cable plugged). We don't know what it might have been used for (and don't need to expose it for now).

Generally trust XboxDevWiki over xbox-linux stuff (we've mostly absorbed their work and improved on it).

Reply:
I'm using the `hdtv` option for true VGA out & digital audio out on Xbox hardware, as a pin configured on the AVIP connector. The `vga` & `vga-sog` options are not officially supported. Sync-on-green is a hack to get `vsync` & `hsync`, but you can tap those signals directly from the GPU, amplify those signals to 5v & not need sync-on-green at all, which is a hardware mod for VGA out.